src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlParameter.cs

```diff
@@ -984,9 +995,11 @@ private void CloneHelper(SqlParameter destination)
 
             destination._metaType = _metaType;
             destination._collation = _collation;
-            destination._xmlSchemaCollectionDatabase = _xmlSchemaCollectionDatabase;
-            destination._xmlSchemaCollectionOwningSchema = _xmlSchemaCollectionOwningSchema;
-            destination._xmlSchemaCollectionName = _xmlSchemaCollectionName;
+            if (_xmlSchemaCollection != null)
+            {
+                destination.EnsureXmlSchemaCollectionExists();
+                destination._xmlSchemaCollection.CopyFrom(_xmlSchemaCollection);
+            }
```

Comment:
After defining `EnsureXmlSchemaCollection` you can replace this if-block with the following line:
```
destination.EnsureXmlSchemaCollection.CopyFrom(_xmlSchemaCollection);
```

Reply:
Done.